conference-sync/ConferenceSync.js

```diff
@@ -0,0 +1,153 @@
+    if (!spreadsheet) {
+        throw new Error('No active spreadsheet, please deploy this script into a spreadsheet context');
+    }
+
+    const sheetName = getScriptProperties_().getProperty(SHEET_NAME_KEY);
+    syncToCalendar_(spreadsheet.getId(), sheetName, calendarId);
+}
+
+/** Synchronize rows from a sheet with well-known columns from a conference form to a calendar. */
+function syncToCalendar_(spreadsheetId, sheetName, calendarId) {
+    const spreadsheet = SpreadsheetApp.openById(spreadsheetId);
+    const calendar = CalendarApp.getCalendarById(calendarId);
+    if (!calendar) {
+        throw new Error(`Failed to access calendar ${calendarId}, please check permissions`);
+    }
+    const sheet = spreadsheet.getSheetByName(sheetName);
+    SheetUtil.mapRows(sheet.getSheetValues(1, 1, -1, -1), null).forEach((row, i) => {
+
+        const untilDeadlineMs = new Date(row.deadline) - new Date();
+        if (!row.notified_deadline_date && untilDeadlineMs) {
+            const deadlineWarnThresholdMs = +(getScriptProperties_().getProperty(DEADLINE_WARNING_DAYS) || 7) * 24 * 60 * 60 * 1000;
+            if (untilDeadlineMs > 0 && untilDeadlineMs < deadlineWarnThresholdMs) {
+                if (notifySlack_(row)) {
+                    sheet.getRange(i + 2, row.$columnIndex.notified_deadline_date + 1).setValue(new Date());
+                }
+            }
+        }
+
+        if (row.calender_url) {
+            // skip entries which have a calendar entry already
+            return;
+        }
+
+        const event = calendar.createAllDayEvent(`${row.conference_name} - CfP Deadline`, row.deadline);
+        event.setLocation(row.submission_website)
+
+        let description = `
+<b>${row.conference_name}</b>
+<u>CfP Closes:</u> ${formatDate_(row.deadline)}
+<u>CfP Submission:</u> ${row.submission_website}`;
+
+        if (row.max_number_of_submissions) {
+            description += `\n<u>Max Submissions:</u> ${row.max_number_of_submissions}`
+        }
+
+        description += "\n"
+
+        if (row.conference_website) {
+            description += `\n<u>Event URL:</u> ${row.conference_website}`
+        }
+        if (row.conference_date_start && row.conference_date_end) {
+            description += `\n<u>Event Dates:</u> ${formatDate_(row.conference_date_start)} - ${formatDate_(row.conference_date_end)}`
+        } else if (row.conference_date_start) {
+            description += `\n<u>Event Date:</u> ${formatDate_(row.conference_date_start)}`
+        }
+        if (row.city || row.country) {
+            description += `\n<u>Location:</u> ${[row.city, row.country].filter(a => !!a).join(', ')}`
+        }
+        if (row.conference_theme) {
+            description += `\n<u>Theme:</u> ${row.conference_theme}`
+        }
+
+        event.setDescription(description);
+
+        // Reminder 7 days before (this only works for the current users reminders array and stacks)
+        // event.addEmailReminder(10080)
```

Comment:
Remove? Or why is it commented out?

Reply:
Complicated story: Event reminders are configured per account on Gcal and didn't work like Marcus imagined them to work. We discussed that already (me and you did so too).

Removing the line now, since Slack notifications have been implemented.